### PR TITLE
[cm_be_47] Add contract for otp login flow

### PR DESCRIPTION
**Pull request by ajayvk-tw**

### CODE REVIEW

File: src/main/resources/static/api.yaml [hunk 1]
```diff
@@ -2172,3 +2242,33 @@ components:
...
+        access_token:
+          type: string
+        expires_in:
+          type: int
+        refresh_expires_in:
+          type: int
+        refresh_token:
+          type: string
+        token_type:
+          type: string
```
Comment (ganesanarun) [inline src/main/resources/static/api.yaml:2274]: Don't we have this already? casing also wrong.

File: src/main/resources/static/api.yaml [hunk 2]
```diff
@@ -2172,3 +2242,33 @@ components:
...
+      properties:
+        username:
+          type: string
+    OtpVerificationResponse:
+      type: object
+      properties:
+        sessionId:
+          type: string
+    OtpPermitRequest:
+      type: object
```
Comment (ganesanarun) [inline src/main/resources/static/api.yaml:2256]: use required attributes mark what are essential.

File: src/main/resources/static/api.yaml [hunk 3]
```diff
@@ -2172,3 +2242,33 @@ components:
...
+    OtpVerificationRequest:
+      type: object
+      properties:
+        username:
+          type: string
+    OtpVerificationResponse:
+      type: object
+      properties:
+        sessionId:
+          type: string
```
Comment (ganesanarun) [inline src/main/resources/static/api.yaml:2254]: Need to send some info about when it's going to expire, to where we send the OTP (mobile/email) with respective value being masked.

Comment (ajayvk-tw) [reply src/main/resources/static/api.yaml:2254]: Where do we get that info from?

File: src/main/resources/static/api.yaml (hunk 1 above)
Comment (ajayvk-tw) [reply src/main/resources/static/api.yaml:2274]: I don't think we have it already, that's why added. which casing? camelcase or lowercase. If camelcase, it's the way current response is.

Comment (ganesanarun) [reply src/main/resources/static/api.yaml:2282]:
```
   AuthResponse:
      type: object
      properties:
        accessToken:
          type: string
          example: eyJhbGciOiJSUzI1Ni.IsInR5cCIgOiAiSldUIiwia2lkIiA6ICJrVVp.2MXJQMjRyYXN1UW9wU2lWbkdZQUZIVFowYVZGVWpYNXFLMnNibTk0In0
        expiresIn:
          type: integer
          example: 1800
          description: In Minutes
        refreshExpiresIn:
          type: integer
          example: 1800
          description: In Minutes
        refreshToken:
          type: string
          example: eyJhbGciOiJSUzI1Ni.IsInR5cCIgOiAiSldUIiwia2lkIiA6ICJrVVp.2MXJQMjRyYXN1UW9wU2lWbkdZQUZIVFowYVZGVWpYNXFLMnNibTk0In0
        tokenType:
          type: string
          example: bearer
```

What's different in your type?

File: src/main/resources/static/api.yaml (hunk 3 above)
Comment (ganesanarun) [reply src/main/resources/static/api.yaml:2256]: Wherever you are going to send OTP, right now it's phone number and OTP expiry is 5 minutes configured in the OTP service.

File: src/main/resources/static/api.yaml (hunk 1 above)
Comment (ajayvk-tw) [reply src/main/resources/static/api.yaml:2282]: camelcasing is not representative of the actual response right..? FTR I didn't see AuthResponse..

File: src/main/resources/static/api.yaml (hunk 2 above)
Comment (ajayvk-tw) [reply src/main/resources/static/api.yaml:2258]: Done.

File: src/main/resources/static/api.yaml (hunk 3 above)
Comment (ajayvk-tw) [reply src/main/resources/static/api.yaml:2256]: Done.

File: src/main/resources/static/api.yaml (hunk 1 above)
Comment (ajayvk-tw) [reply src/main/resources/static/api.yaml]: Removed redundant addition

Comment (ganesanarun) [reply src/main/resources/static/api.yaml]: our response should always be in camelcase.

File: src/main/resources/static/api.yaml [hunk 4]
```diff
@@ -1208,6 +1208,76 @@ paths:
...
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AuthResponse'
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/AuthResponse'
+        401:
+          description: Unauthorized
+        400:
+          description: Bad request(potentially bad request token)
```
Comment (ganesanarun) [inline src/main/resources/static/api.yaml:1271]: What exactly you mean by that?

File: src/main/resources/static/api.yaml (hunk 2 above)
Comment (ganesanarun) [reply src/main/resources/static/api.yaml:2269]: Do we need username as part of the request?

File: src/main/resources/static/api.yaml (hunk 4 above)
Comment (ajayvk-tw) [reply src/main/resources/static/api.yaml:1271]: I think it's incorrectly copied, will remove the 400 error section.

File: src/main/resources/static/api.yaml (hunk 1 above)
Comment (ajayvk-tw) [reply src/main/resources/static/api.yaml]: Yes. I've updated the api to use the existing AuthResponse only.

**Pull request merged**